lib/bundler/source/rubygems.rb

```diff
@@ -302,6 +302,7 @@ def remote_specs
           # the gemspecs of those gems, if the non-api sites contain more than
           # about 100 gems, we just treat all sites as non-api for speed.
           allow_api = idx.size < API_REQUEST_LIMIT && dependency_names.size < API_REQUEST_LIMIT
+          Bundler.ui.debug "Need to query more then #{API_REQUEST_LIMIT} gems" unless allow_api
```

Comment:
I like the second one better. Cause *and* effect.

Also, could you change `then` to `than`? Thank you.